File: bugs/core/world/town.py
from .entity import Entity
from .entity_types import EntityTypes
from .size import Size

class Town(Entity):

    def __init__(self, main_event_bus, id, pos, color):
        super().__init__(main_event_bus, EntityTypes.TOWN, id, pos, Size(100, 100))
        self._color = color

    def get_color(self):
        return self._color

    def to_json(self):
        json = super().to_json()
        json.update({
            'color': self._color
        })
        return json

File: bugs/core/client/src/domain/entity/town.js
import { Entity } from './entity';
import { EntityTypes } from './entityTypes';

class Town extends Entity {

    constructor(eventBus, id, position, color) {
        super(eventBus, id, position, EntityTypes.TOWN);
        this._color = color
    }

    get color() {
        return this._color
    }

}

export {
    Town
}
